sae/blocks.go

```diff
@@ -174,23 +173,7 @@ func (vm *VM) buildBlock(
 		}
 	}
 
-	// TODO: This header hasn't had an opportunity to be populated by,
-	// [hook.Points] but it is passed into [hook.Points.SubSecondBlockTime]. We
-	// should construct the header inside of [hook.Points].
-	hdr := &types.Header{
-		ParentHash: parent.Hash(),
-		Root:       lastSettled.PostExecutionStateRoot(),
-		Number:     new(big.Int).Add(parent.Number(), common.Big1),
-		// GasLimit is populated after [worstcase.State.StartBlock].
-		// GasUsed should be populated after [worstcase.State.ApplyTx] has
-		// finished being called.
-		Time: unix(blockTime),
-		// BaseFee is populated after [worstcase.State.StartBlock].
-
-		// All other fields are, optionally, populated by the call to
-		// [hook.BlockBuilder.BuildBlock] after applying every
-		// [types.Transaction].
-	}
+	hdr.Root = lastSettled.PostExecutionStateRoot()
```

Comment:
I think we should log warnings if any of the fields that are SAE's sole responsibility are already populated. That'll result in failed tests (if using `saetest.TBLogger`) for early detection and also save a lot of pain-in-the-arse debugging for someone who erroneously populates them.